File: mobile/lib/data/models/conversation_model.dart
class ConversationModel {
  const ConversationModel({
    required this.id,
    required this.buyerId,
    required this.sellerId,
    required this.productId,
    required this.createdAt,
    this.productTitle,
    this.productImage,
    this.buyerEmail,
    this.sellerEmail,
  });

  final String id;
  final String buyerId;
  final String sellerId;
  final String productId;
  final DateTime createdAt;
  final String? productTitle;
  final String? productImage;
  final String? buyerEmail;
  final String? sellerEmail;

  factory ConversationModel.fromJson(Map<String, dynamic> json) {
    return ConversationModel(
      id: json['id'] as String,
      buyerId: json['buyer_id'] as String,
      sellerId: json['seller_id'] as String,
      productId: json['product_id'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      productTitle: json['product_title'] as String?,
      productImage: json['product_image'] as String?,
      buyerEmail: json['buyer_email'] as String?,
      sellerEmail: json['seller_email'] as String?,
    );
  }

  String getDisplayTitle(String currentUserId) {
    if (currentUserId == buyerId) {
      return sellerEmail ?? 'Satıcı';
    }
    return buyerEmail ?? 'Alıcı';
  }
}

File: src/models/conversation.model.js
const { query } = require("../config/database");

const CONV_SELECT = "id, buyer_id, seller_id, product_id, created_at";

function run(client, text, params) {
  return client ? client.query(text, params) : query(text, params);
}

async function findConversationByParticipants(buyerId, sellerId, productId, client) {
  const r = await run(
    client,
    "SELECT " + CONV_SELECT + " FROM conversations WHERE buyer_id = $1 AND seller_id = $2 AND product_id = $3",
    [buyerId, sellerId, productId]
  );
  return r.rows[0] || null;
}

async function insertConversation({ buyerId, sellerId, productId }, client) {
  const r = await run(
    client,
    "INSERT INTO conversations (buyer_id, seller_id, product_id) VALUES ($1, $2, $3) RETURNING " + CONV_SELECT,
    [buyerId, sellerId, productId]
  );
  return r.rows[0];
}

async function findUserConversations(userId) {
  const r = await query(
    `SELECT c.*, 
            p.title as product_title, p.image_url as product_image,
            ub.email as buyer_email, us.email as seller_email
     FROM conversations c
     JOIN products p ON c.product_id = p.id
     JOIN users ub ON c.buyer_id = ub.id
     JOIN users us ON c.seller_id = us.id
     WHERE c.buyer_id = $1 OR c.seller_id = $1
     ORDER BY c.created_at DESC`,
    [userId]
  );
  return r.rows;
}

async function findConversationById(id) {
  const r = await query("SELECT " + CONV_SELECT + " FROM conversations WHERE id = $1", [id]);
  return r.rows[0] || null;
}

module.exports = {
  findConversationByParticipants,
  insertConversation,
  findUserConversations,
  findConversationById,
};
